src/gui/syncconflictsmodel.cpp

```diff
@@ -0,0 +1,355 @@
+        Q_EMIT allExistingsSelectedChanged();
+    }
+}
+
+void SyncConflictsModel::selectAllConflicting(bool selected)
+{
+    for (auto &singleConflict : mConflictData) {
+        singleConflict.mConflictSelected = selected;
+    }
+
+    Q_EMIT dataChanged(index(0), index(rowCount() - 1), {static_cast<int>(SyncConflictRoles::ConflictSelected)});
+
+    if (selected && !mAllConflictingsSelected) {
+        mAllConflictingsSelected = true;
+        Q_EMIT allConflictingSelectedChanged();
+    } else if (!selected && mAllConflictingsSelected) {
+        mAllConflictingsSelected = false;
+        Q_EMIT allConflictingSelectedChanged();
+    }
+}
+
+void SyncConflictsModel::applyResolution()
+{
+    for(const auto &syncConflict : qAsConst(mConflictData)) {
+        if (syncConflict.isValid()) {
+            qCInfo(lcSyncConflictsModel) << syncConflict.mExistingFilePath << syncConflict.mConflictingFilePath << syncConflict.solution();
+            ConflictSolver solver;
+            solver.setLocalVersionFilename(syncConflict.mConflictingFilePath);
+            solver.setRemoteVersionFilename(syncConflict.mExistingFilePath);
+            solver.exec(syncConflict.solution());
+        }
+    }
+}
+
+void SyncConflictsModel::updateConflictsData()
+{
+    mConflictData.clear();
+    mConflictData.reserve(mData.size());
+
+    for (const auto &oneConflict : qAsConst(mData)) {
+        if (!FolderMan::instance()) {
+            qCWarning(lcSyncConflictsModel) << "no FolderMan instance";
+            mConflictData.push_back({});
+            continue;
+        }
+        const auto folder = FolderMan::instance()->folder(oneConflict._folder);
+        if (!folder) {
+            qCWarning(lcSyncConflictsModel) << "no Folder instance for" << oneConflict._folder;
+            mConflictData.push_back({});
+            continue;
+        }
+
+        const auto conflictedRelativePath = oneConflict._file;
+        const auto dbRecord = folder->journalDb();
+        const auto baseRelativePath = dbRecord ? dbRecord->conflictFileBaseName(conflictedRelativePath.toUtf8()) : QString{};
+
+        const auto dir = QDir(folder->path());
+        const auto conflictedPath = dir.filePath(conflictedRelativePath);
+        const auto basePath = dir.filePath(baseRelativePath);
+
+        const auto existingFileInfo = QFileInfo(basePath);
+        const auto conflictFileInfo = QFileInfo(conflictedPath);
+
+        auto newConflictData = ConflictInfo{
+            existingFileInfo.fileName(),
+            mLocale.formattedDataSize(existingFileInfo.size()),
+            mLocale.formattedDataSize(conflictFileInfo.size()),
+            existingFileInfo.lastModified().toString(),
+            conflictFileInfo.lastModified().toString(),
+            QUrl{QStringLiteral("image://tray-image-provider/:/fileicon") + existingFileInfo.filePath()},
+            QUrl{QStringLiteral("image://tray-image-provider/:/fileicon") + conflictFileInfo.filePath()},
+            false,
```

Comment:
@mgallien it is hard to read these bools, `enum` may help

Reply:
done